lumicks/pylake/recording.py

```diff
@@ -0,0 +1,154 @@
+
+    @property
+    def stop(self):
+        timestamp_string = re.search('^\d+:(\d+)$', self._src.tags['DateTime'].value)
+        return np.int64(timestamp_string.group(1)) if timestamp_string else None
+
+
+class TiffStack:
+    """TIFF images exported from Bluelake
+
+        Parameters
+        ----------
+        tiff_file : tifffile.TiffFile
+            Filename pointing to a TIFF file recorded from a camera in Bluelake.
+        """
+    def __init__(self, tiff_file):
+        self._src = tiff_file
+
+    def get_frame(self, frame):
+        return TiffFrame(self._src.pages[frame])
+
+    @staticmethod
+    def from_file(image_file):
+        import tifffile
+        return TiffStack(tifffile.TiffFile(image_file))
+
+    @property
+    def num_frames(self):
+        return len(self._src.pages)
+
+
+class Recording:
+    """Recording obtained with Bluelake
+
+        Parameters
+        ----------
+        data : TiffStack
+            TiffStack object.
+        name : str
+            Plot label of the recording
+        start_idx : int
+            Index at the first frame.
+        stop_idx: int
+            Index beyond the last frame.
+        """
+    def __init__(self, data, name=None, start_idx=0, stop_idx=None):
+        self.src = data
+        self.name = name
+        self.start_idx = start_idx
+        self.stop_idx = (self.src.num_frames if stop_idx is None else stop_idx)
+
+    def __getitem__(self, item):
+        """All indexing is in frames"""
+        if isinstance(item, slice):
+            if item.step is not None:
+                raise IndexError("Slice steps are not supported")
+
+            start, stop, _ = item.indices(self.num_frames)
+            return Recording(self.src, self.name, self.start_idx + start, self.start_idx + stop)
+        else:
+            item = self.start_idx + item if item >= 0 else self.stop_idx + item
+            if item >= self.stop_idx or item < self.start_idx:
+                raise IndexError("Index out of bounds")
+            return Recording(self.src, self.name, item, item + 1)
+
+    @staticmethod
+    def from_file(image_name):
+        """Construct recording from file
+
+        Parameters
+        ----------
+        image_name : str
+            Filename for the image stack.
+        """
+        return Recording(TiffStack.from_file(image_name), os.path.splitext(os.path.basename(image_name))[0])
+
+    def plot(self,  **kwargs):
+        import matplotlib.pyplot as plt
+
+        default_kwargs = dict(
+            cmap='gray'
+        )
+
+        frame = np.clip(kwargs.pop("frame", 0), 0, self.num_frames)
+        image = self.get_frame(frame).data
+        plt.imshow(image, **{**default_kwargs, **kwargs})
+
+        if self.num_frames == 1:
+            plt.title(self.name)
+        else:
+            plt.title(f"{self.name} [frame {frame}/{self.num_frames}]")
+
+    def get_frame(self, frame=0):
```

Comment:
Agreed. I added a raw function which returns a list of raw frame objects, or just a single raw frame and made _get_frame private, since I would still prefer this for internal class use.

recording[n].raw.start will now give you the start position of that frame.
recording[n].raw.data will give you the raw pixel data.